### PR TITLE
Add position filters to leaders

### PR DESCRIPTION
**Pull request by kgilles**
This adds two new filters to the Leaders page: Forwards & Defensemen

**Desktop**
image

**Mobile**
image

### CODE REVIEW

File: components/Selector/LeadersFilterSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+  useEffect(() => {
+    if (isExpanded && selectorRef.current) {
+      selectorRef.current.addEventListener('mouseleave', onMouseLeave);
+    }
+  }, [selectorRef, isExpanded]);
+
+  useEffect(() => setSelectedLeadersFilter(activeFilter), [activeFilter]);
+
+  const onButtonClick = () => setIsExpanded(!isExpanded);
```
Comment (esilverm) [inline components/Selector/LeadersFilterSelector.tsx]:
```suggestion
  const onButtonClick = useCallback(() => setIsExpanded((expanded) => !expanded), [setIsExpanded]);
```
Let's start trying to incorporate some more `useCallbacks` here. Make sure that this works properly first though

File: hooks/useLeaders.ts [hunk 2]
```diff
@@ -39,7 +40,7 @@ const useLeaders = (
   const endpoint =
     playerType.toLowerCase() === 'goalie'
       ? `/goalies/${stat}`
-      : `/skaters/${stat}`;
+      : `/skaters/${stat}?position=${position}&`;
```
Comment (esilverm) [inline hooks/useLeaders.ts]: Nit:
```suggestion
      : `/skaters/${stat}?position=${position}`;
```

File: pages/[league]/leaders.tsx [hunk 3]
```diff
@@ -66,11 +61,14 @@ function Stats({ league }: Props): JSX.Element {
   }, []);
 
   const onSeasonTypeSelect = (type) => setSeasonType(type);
```
Comment (esilverm) [inline pages/[league]/leaders.tsx]:
```suggestion
  const onSeasonTypeSelect = useCallback((type) => setSeasonType(type), [setSeasonType]);
```

File: pages/[league]/leaders.tsx [hunk 4]
```diff
@@ -66,11 +61,14 @@ function Stats({ league }: Props): JSX.Element {
...
+  const onLeadersFilterSelect = (filter) => setFilter(filter);
```
Comment (esilverm) [inline pages/[league]/leaders.tsx]:
```suggestion
  const onLeadersFilterSelect = useCallback((filter) => setFilter(filter), [setFilter]);
```

**Review by esilverm — APPROVED** [review]
A few potential changes to hopefully make things more efficient in the future (and lower my workload when I overhaul this code sometime). I'll merge after changes

File: hooks/useLeaders.ts (hunk 2 above)
Comment (kgilles) [reply hooks/useLeaders.ts]: Good catch. I'll move this out of the endpoint and create a `positionParam` instead. It doesn't break anything if it's sent to the goalie leaders endpoint.

File: components/Selector/LeadersFilterSelector.tsx (hunk 1 above)
Comment (kgilles) [reply components/Selector/LeadersFilterSelector.tsx]: I've made changes to use `useCallback` where suggested. Tested them as well and everything still works.

I'll keep in mind to start using this consistently from now on.

Comment (esilverm) [reply components/Selector/LeadersFilterSelector.tsx]: Awesome! Thanks! 

These usually are helpful with handlers and will reduce re-renders

**Pull request merged**